src/Combinators/orGuard.ts

```diff
@@ -0,0 +1,45 @@
+import {ReasonGuard} from '../ReasonGuard';
+import {andGuard} from './thenGuard';
+import {notGuard} from './notGuard';
+import {buildNegatable} from '../NegatableGuard';
+
+export const orGuard = <FROM, LEFT extends FROM, RIGHT extends FROM>(
+	left: ReasonGuard<FROM, LEFT>,
+	right: ReasonGuard<FROM, RIGHT>
+) => buildNegatable(
+		() => getRawOr(left, right),
+		() => getRawNegatedOr(left, right)
+	);
+
+function getRawOr<FROM, LEFT extends FROM, RIGHT extends FROM>(
+	left: ReasonGuard<FROM, LEFT>,
+	right: ReasonGuard<FROM, RIGHT>
+): ReasonGuard<FROM, LEFT|RIGHT> {
+	return (input, output = [], confirmations = []): input is LEFT|RIGHT => {
+		try {
+			const innerErrors: Error[] = [];
+			const innerConfs: string[] = [];
+			if (left(input, innerErrors, innerConfs)) {
+				confirmations.push(...innerConfs);
+				return true;
+			}
+			innerErrors.splice(1);
```

Comment:
Why assume it will only return a single error?  The only reason I see is to ease the error string building below?

Reply:
It's to avoid the need to build up a hierarchy of messages and keep things string-y. I do have some thoughts on moving to a hierarchy or other model for rguards to give feedback, but this is it for now. Note that having the or itself provide multiple errors would be inherently weird in the current setup, which is what forces further awkwardness.